lib/player.rb

```diff
@@ -0,0 +1,87 @@
+require_relative '../lib/tilebag'
+
+module Scrabble
+  class Player
+    attr_reader :name, :plays, :tiles
+    def initialize(name, debug = false)
+      @name = name
+      @plays = []
+      @score_array = []
+      @tiles = []
+      @debug = debug
+    end#initialize
+
+    def play(word)
+      if won?
+        false
+      else
+        @plays << word.upcase
+        delete_tiles_from_tiles_array(word)
+        word_score = Scrabble::Scoring.score(word)
+        @score_array << word_score
+        return word_score
+      end
+    end #play word end
+
+    def delete_tiles_from_tiles_array(word)
+      word = word.upcase
+      word.split("").each do |letter|
+        if @tiles.include?(letter)
+          @tiles.slice!(@tiles.index(letter))
+        else
+          raise ArgumentError.new("You cannot play this word.")
```

Comment:
While this appears to be doing what we want, it's actually got some subtly incorrect behavior. Here's a test case that might expose what is going wrong:

```ruby
it "doesn't remove player's tiles if the word cannot be played" do
  jugador = Scrabble::Player.new("Morsa", true)
  jugador.draw_tiles(@tile_bag, ["C", "A", "C", "K", "L", "E", "S"])

  proc { jugador.play("CACKLED") }.must_raise ArgumentError
  jugador.play("CACKLES").must_equal 15
end
```

Try adding that to the test suite for `Player#play` and then figure out what changes are needed to make it pass.